ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx

```diff
@@ -15,24 +15,250 @@
+import { ContentTypeField } from '../../../models/ContentType';
+import { nnou, nou, reversePluckProps } from '../../../utils/object';
+import GlobalState from '../../../models/GlobalState';
+import ContentInstance from '../../../models/ContentInstance';
+import Input from '../../../components/Controls/FormEngine/Input';
+import Dropdown from '../../../components/Controls/FormEngine/Dropdown';
+import CheckboxGroup from '../../../components/Controls/FormEngine/CheckboxGroup';
+import DateTime from '../../../components/Controls/FormEngine/DateTime';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    fab: {
+      margin: theme.spacing(1),
+    },
+    formControl: {
+      width: '100%',
+      '& .MuiFormGroup-root': {
+        marginLeft: '10px',
+      },
+      '& .MuiInputBase-root': {
+        marginTop: '12px !important',
+      }
+    },
+    ActionButton: {
+      padding: '6px',
+      marginLeft: '5px'
+    },
+    InputLabel: {
+      position: 'relative'
+    },
+    PanelMargin: {
+      margin: '0 15px',
+    },
+    textField: {
+      width: '100%',
+    },
+    actionBTN:{
+      margin:'18px 0 80px 15px',
+      '& .MuiButton-contained':{
+        marginRight: '8px',
+      }
+    },
+    divider: {
+      margin: '23px 0 10px',
+    }
+  }),
+);
 
 const translations = defineMessages({
   audiencesPanel: {
     id: 'craftercms.ice.audiences.title',
     defaultMessage: 'Audience Targeting'
+  },
+  audiencesPanelLoading: {
+    id: 'craftercms.ice.audiences.loading',
+    defaultMessage: 'Retrieving targeting options'
   }
 });
 
-export default function AudiencesPanel() {
+const controlsMap = {
+  'dropdown': Dropdown,
+  'checkbox-group': CheckboxGroup,
+  'date-time': DateTime,
+  'input': Input
+};
+
+interface AudiencesPanelUIProps {
+  audiencesResource: any;
+  model: ContentInstance;
+  modelApplying: boolean;
+  modelApplied: boolean;
+  onFormChange: Function;
+  onSaveModel: Function;
+  onSetDefaults: Function;
+}
+
+export function AudiencesPanelUI(props: AudiencesPanelUIProps) {
+
+  const classes = useStyles({});
+  const {
+    audiencesResource,
+    model,
+    modelApplying,
+    onFormChange,
+    onSaveModel,
+    onSetDefaults
+  } = props;
+  const contentType = audiencesResource.read();
+
   return (
     <ToolPanel title={translations.audiencesPanel}>
-      <Typography component="h2" variant="subtitle1" style={{ padding: '10px' }}>
-        Audiences Panel
-      </Typography>
+      {
+        <>
+          <Grid className={classes.PanelMargin}>
+            {
+              Object.keys(contentType.fields).map((field: any) => {
+                const type = contentType.fields[field].type;
+                const Control = controlsMap[type];
+
+                const controlProps = {
+                  field: contentType.fields[field],
+                  value: model[field] ? model[field].key : undefined,
```

Comment:
Why reading key? Inside the model, values should be directly under the [fieldId] prop.